trainer.py

```diff
@@ -85,28 +105,29 @@ def train(model_actor, random_actor, gamma, num_episodes, optimizer):
             prob_action, _ = sample
             current_discounted_return = discounted_returns[i]
             # TODO: Can we somehow reduce # of invalid samples generated during battle?
-            # prob_action is None when it is not p1's move so maybe it is okay to ignore these cases? 
+            # prob_action is None when it is not p1's move so maybe it is okay to ignore these cases?
             # It is interesting that this number fluctuates greatly though!
             if (prob_action is None):
                 invalid_samples += 1
                 continue
 
-            # Important Observations: 
+            # Important Observations:
             # (1) High loss values could mean prob_action is close to 0 often (poor model convergence?)
             # (2) Hacky fix for now is to add 1e-10 to prob_action to avoid taking log of 0.
             loss = -torch.log(prob_action + 1e-10) * current_discounted_return
 
-            total_loss += loss # negative loss values are due to negative rewards
+            total_loss = total_loss + loss  # negative loss values are due to negative rewards
 
         # This check is done as sometimes training samples are too few / invalid causing total_loss to be 0
-        if (total_loss != 0):
+        if (total_loss != 0 and not torch.isnan(total_loss)):
```

Comment:
One thing I have noticed in other implementations of loss is taking the mean over all samples before backpropagating. I can test it out once you merge, or if you want to make this quick change now, feel free!

Reply:
Got it, I can try adding that